disperser/controller/dispatcher.go

```diff
@@ -0,0 +1,432 @@
+					if err != nil {
+						d.logger.Error("failed to handle signatures", "err", err)
+					}
+				}()
+			}
+		}
+	}()
+
+	return nil
+
+}
+
+func (d *Dispatcher) HandleBatch(ctx context.Context) (chan core.SigningMessage, *batchData, error) {
+	currentBlockNumber, err := d.chainState.GetCurrentBlockNumber()
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get current block number: %w", err)
+	}
+	referenceBlockNumber := uint64(currentBlockNumber) - d.FinalizationBlockDelay
+
+	// Get a batch of blobs to dispatch
+	// This also writes a batch header and blob verification info for each blob in metadata store
+	batchData, err := d.NewBatch(ctx, referenceBlockNumber)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	batch := batchData.Batch
+	state := batchData.OperatorState
+	sigChan := make(chan core.SigningMessage, len(state.IndexedOperators))
+	for opID, op := range state.IndexedOperators {
+		opID := opID
+		op := op
+		client, err := d.nodeClientManager.GetClient(opID, op.Socket)
+		if err != nil {
+			d.logger.Error("failed to get node client", "operator", opID, "err", err)
+			continue
+		}
+
+		d.pool.Submit(func() {
+
+			req := &corev2.DispersalRequest{
+				OperatorID: opID,
+				// TODO: get OperatorAddress
+				OperatorAddress: gethcommon.Address{},
+				Socket:          op.Socket,
+				DispersedAt:     uint64(time.Now().UnixNano()),
+				BatchHeader:     *batch.BatchHeader,
+			}
+			err := d.blobMetadataStore.PutDispersalRequest(ctx, req)
+			if err != nil {
+				d.logger.Error("failed to put dispersal request", "err", err)
+				return
+			}
+
+			for i := 0; i < d.NumRequestRetries+1; i++ {
+				sig, err := d.sendChunks(ctx, client, batch)
+				if err == nil {
+					storeErr := d.blobMetadataStore.PutDispersalResponse(ctx, &corev2.DispersalResponse{
+						DispersalRequest: req,
+						RespondedAt:      uint64(time.Now().UnixNano()),
+						Signature:        sig.Bytes(),
+						Error:            "",
+					})
+					if storeErr != nil {
+						d.logger.Error("failed to put dispersal response", "err", storeErr)
+					}
+
+					sigChan <- core.SigningMessage{
+						Signature:            sig,
+						Operator:             opID,
+						BatchHeaderHash:      batchData.BatchHeaderHash,
+						AttestationLatencyMs: 0, // TODO: calculate latency
+						Err:                  nil,
+					}
+
+					break
+				}
+
+				d.logger.Warn("failed to send chunks", "operator", opID, "NumAttempts", i, "err", err)
+				time.Sleep(time.Duration(math.Pow(2, float64(i))) * time.Second) // Wait before retrying
+			}
+		})
+	}
+
+	return sigChan, batchData, nil
+}
+
+// HandleSignatures receives signatures from operators, validates, and aggregates them
+func (d *Dispatcher) HandleSignatures(ctx context.Context, batchData *batchData, sigChan chan core.SigningMessage) error {
+	quorumAttestation, err := d.aggregator.ReceiveSignatures(ctx, batchData.OperatorState, batchData.BatchHeaderHash, sigChan)
```

Comment:
Is there any risk of receiving signatures from the same operator twice or more times ? Even if we do, is it a problem ?

Reply:
No. Each request should [result](https://github.com/Layr-Labs/eigenda/pull/871/files/469b084bd08f16502f6037b5260eff7103a9747e#diff-96cf41ed19eafb59625ea2cc3208dd5cc930d68d6f64a278dfd2783a620b98a9R135) in one (or no) response back to this channel